File: 알고리즘/1/이진 변환 반복하기.py
from typing import Counter


def solution(s):
    answer = []
    zero = 0
    i = 0
    while True:
        i+=1
        zero+=s.count('0')
        s= bin(s.count('1')).replace('0b','')
        if(s=='1'):
            break
        
    return [i, zero]

File: algorithms/1/이진 변환 반복하기.cpp
#include <bits/stdc++.h>

using namespace std;

string n_to_2(int one){
    string bin;
    while(one>0){
        bin = to_string(one%2)+bin;
        one/=2;
    }
    return bin;
}

vector<int> solution(string s) {
    vector<int> answer;
    
    int i=0;
    int zero = 0;
    while(true){
        i+=1;
        zero+=count(s.begin(), s.end(), '0');
        int one = count(s.begin(), s.end(), '1');

        s = n_to_2(one);
        
        if(s=="1") break;
    }
    answer.push_back(i);
    answer.push_back(zero);
    return answer;
}